### PR TITLE
Convert install.md text into install.sh script

### PR DESCRIPTION
**Pull request by afbjorklund**
Try to make an executable version of the install:

https://github.com/cri-o/cri-o/blob/master/install.md#install-packaged-versions-of-cri-o

Use `sudo` for getting the required root access

Quote things at random for shellcheck/shfmt...

For #4343

### CODE REVIEW

**Comment by afbjorklund** [comment]
After the first pathetic attempts, it seems to work OK in centos and ubuntu containers (after first installing `curl` and `gnupg`)

However, not all versions of cri-o are available for operating systems - so maybe $VERSION should be a parameter ($1) ?

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
+#!/bin/sh -ex
```
Comment (saschagrunert) [inline install.sh:1]: Let's link this script in the readme/install instructions.

**Comment by haircommander** [comment]
> After the first pathetic attempts, it seems to work OK in centos and ubuntu containers (after first installing `curl` and `gnupg`)
> 
> However, not all versions of cri-o are available for operating systems - so maybe $VERSION should be a parameter ($1) ?

yeah I definitely think $VERSION should be a parameter

**Comment by haircommander** [comment]
thanks for picking this up @afbjorklund !

**Comment by afbjorklund** [comment]
> yeah I definitely think $VERSION should be a parameter

Done.

**Comment by haircommander** [comment]
/retest
/approve
LGTM, there's one outstanding nit about adding a pointer from the README and install docs

**Comment by afbjorklund** [comment]
> LGTM, there's one outstanding nit about adding a pointer from the README and install docs

Were you able to find anywhere to host it ? Or would it have to use some "raw" github URL ?

**Comment by haircommander** [comment]
> > LGTM, there's one outstanding nit about adding a pointer from the README and install docs
> 
> Were you able to find anywhere to host it ? Or would it have to use some "raw" github URL ?
> 
> Like https://get.docker.com

for starters, it can be self referential. like if we're updating one of the readme files, we can put a link to `(/install.sh)`

**Comment by afbjorklund** [comment]
I think GitHub will add the usual frame to it, if you just link it in the markdown ?

Like so:[ install.sh](https://github.com/afbjorklund/cri-o/blob/install/install.sh) 

So you would need something like:

`curl -sSL https://raw.githubusercontent.com/afbjorklund/cri-o/install/install.sh | sh -`

**Comment by haircommander** [comment]
I think we should just have a reference to it for now:
```
You can install from package managers with our [install script](/install.md), or by following the steps below:
```

after the title here https://github.com/cri-o/cri-o/blob/master/install.md#install-packaged-versions-of-cri-o

**Comment by afbjorklund** [comment]
> I think we should just have a reference to it

What would be the use of having a _second_ install.md ?

The whole idea was to have something that was executable

**Comment by haircommander** [comment]
haha I think I meant
```
You can install from package managers with our [install script](/install.sh), or by following the steps below:
```

**Comment by haircommander** [comment]
thinking about this more, I Think we can add references to docs as a follow up

PTAL @umohnani8 @kolyshkin @saschagrunert @mrunalp

**Comment by afbjorklund** [comment]
> haha I think I meant
> 
> ```
> You can install from package managers with our [install script](/install.sh), or by following the steps below:
> ```

It's still just a link to a web page, though.

**Comment by afbjorklund** [comment]
Ultimately we would like something (an URL) that can be fed into the "Provisioner" of the libmachine replacement.

The role of the provisioner is to install the container runtime on the node, for later bootstrapping of Kubernetes...

Right now it is hardcoded to Docker, and only allows different _versions_ of it.

But we want to have equal support for docker and containerd - and also cri-o.

**Comment by haircommander** [comment]
we can add it here https://cri-o.github.io/cri-o/ under the CRI-O tab, above the release-notes. I am not super sure how to do that though, and @saschagrunert is still out on vacation I believe

**Comment by TomSweeneyRedHat** [comment]
The script itself LGTM.

I'm not sure if this is being suggested or not, but I would not completely replace install.md with this script.  For instance, there's a lot of good info on how to install/setup your sandbox in the install.md that I'd not want to lose.

**Pull request closed without merging**